app/services/external_apis/base_service.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+module ExternalApis
+
+  class ExternalApiError < StandardError; end
+
+  class BaseService
```

Comment:
I really like this base-API-service approach.  Particularly with the maDMP need to consume other APIs for the various controlled vocabularies/identifiers.